control/trajectory_follower/src/pid_longitudinal_controller.cpp

```diff
@@ -539,45 +554,70 @@ PidLongitudinalController::ControlState PidLongitudinalController::updateControl
           control_data.stop_dist -
           0.5 * (pred_vel_in_target + current_vel) * m_delay_compensation_time;
         m_smooth_stop.init(pred_vel_in_target, pred_stop_dist);
-        return ControlState::STOPPING;
+        return print_and_return(ControlState::STOPPING);
       }
     } else {
       if (stopped_condition && !departure_condition_from_stopped) {
-        return ControlState::STOPPED;
+        return print_and_return(ControlState::STOPPED);
       }
     }
-  } else if (current_control_state == ControlState::STOPPING) {
+    return print_and_return(ControlState::DRIVE);
```

Comment:
This seems always called in the drive state.

IMO, better to have prev_control_state_ and compare it with current one.

Reply:
change implementation. Please check it.